docs/DeveloperGuide.md

```diff
@@ -167,13 +167,51 @@ Classes used by multiple components are in the `seedu.taassist.commons` package.
 
 ## **Implementation**
 
+### Creating and deleting classes
+
+Class information is stored as `ModuleClass` objects, which captures the name of the class as well as the sessions created for the class. When the user creates a class, the program creates a new `ModuleClass` object with the class name and adds it to the collection of `ModuleClass` objects previously created, which is managed by the `TaAssist` class.
+
+When deleting classes, all students previously assigned to those classes will have the classes unassigned from them, before the `ModuleClass` objects corresponding to those classes are removed from the collection captured by `TaAssist`.
+
+The following methods in `TaAssist` manages the adding and deleting of classes from the collection:
+
+- `TaAssist#addModuleClass(ModuleClass moduleClass)` - Adds the provided class to the list of classes created.
+- `TaAssist#removeModuleClass(ModuleClass moduleClass)` - Removes the provided class from the list of classes created.
 
 ### Assigning students to a class
 <img src="images/AssignCommandSequenceDiagram.png" width="700" />
 
-DESCRIPTION TODO
+Each student object contains a collection of `StudentModuleData` where classes and the grades the student obtained for the sessions of the classes are stored. When the user tries to assign students to a class, we are essentially creating a new `StudentModuleData` object to be added to the collection for each student.
+
+Given below are the different steps taken that when the user tries to assign students to a class.
+
+Step 1: The user enters the command keyword `assign`, followed by the indices of the students he want to assign classes to, and the names of the classes that he wants to assign to the students. Example: `assign 1 2 3 c/CS1101S`.
+
+Step 2: The program makes use of the `TAAssistParser` to make sense of the keyword, and determine which parser to use to parse the arguments. In this case, the `AssignCommandParser` will be used.
```

Comment:
I think replacing instances of `the program` with `ta assist` will match the context better.

Reply:
I am afraid that might create confusion with the `TaAssist` class within the program. :(

Reply:
I feel that it's fine as long as we don't make it a code block (i.e. "TA Assist" vs `taAssist`). Anyway I think this can be left later when we standardise everything